tests/typing/simple_if_null_check.sy

```diff
@@ -0,0 +1,22 @@
+f :: fn a: bool -> int?
+    if a do
+        ret 1
+    else do
+        ret nil
+    end
+end
+
+start :: fn do
+    a := f(true)
+
+    if :int is a do
+        a = a + 1
+    end
```

Comment:
Does this work for blobs as well?

Reply:
If you mean blob field accesses, no. If you mean union types with blobs, yes it should work.